iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/load/splitter/TsFileSplitter.java

```diff
@@ -389,8 +389,20 @@ private boolean checkMagic(TsFileSequenceReader reader) throws IOException {
     }
 
     byte versionNumber = reader.readVersionNumber();
-    if (versionNumber != TSFileConfig.VERSION_NUMBER) {
-      logger.error("the file's Version Number is incorrect, file path: {}", reader.getFileName());
+    if (versionNumber < TSFileConfig.VERSION_NUMBER) {
+      if (versionNumber < TSFileConfig.VERSION_NUMBER - 1) {
+        logger.error("the file's Version Number is too old, file path: {}", reader.getFileName());
+        return false;
+      } else {
+        logger.warn(
```

Comment:
```suggestion
        logger.info(
```

Reply:
Need to test loading v3 files into v4 IoTDB with type dismatch conditions.

Reply:
fixed